feat: add release date and modules version

HinataKah0 opened the pull request:
Hi @bnb ,

Background:
We are currently consolidating the Node releases data logic/code in `nodejs/nodejs.org` repo (I am referencing this [PR](https://github.com/nodejs/nodejs.org/pull/5365)). And the repo has been using `nodevu` for some time. Then we found out that there are some fields that we need but not returned by `nodevu` yet.

We need these 2 new fields:
- `date` -> fetched from `index.json` and it's not the same with `support.phases.dates.start`. I am not too sure myself about the historical context behind it. However, it seems like: release date in `index.json` is referring to actual the release date and `support.phases.dates.start` is referring to the support start date ("current" status starts).
- `modules.version` -> fetched from `index.json` as well, see [this](https://nodejs.org/en/download/releases#backref-1) for explanation.

This PR is still in draft state (I haven't added any test or added to the docs), looking forward for your inputs/feedbacks.

HinataKah0 committed:
feat: add release date and modules version

bnb commented:
I'd ignore the lts-2 CI - I believe that release doesn't actually support fetch in Node.js.

Would you be able to add tests for the new features? :)

bnb commented on core/index.js:
```diff
@@ -45,6 +45,14 @@ async function core (options) {
 
     data[name].releases[`v${versionSemver.version}`].semver = semverToReturn
 
+    data[name].releases[`v${versionSemver.version}`].date = versions[version].date ?? undefined
```
I've gotten to dig in a bit now after fixing up my personal laptop setup. I do have one request here, would it be possible to name this `release` or `releaseDate` rather than `date`? My goal with this module is specifically to make it so the code can be a readable API without having to refer to docs or having to already have experience with `dist/index.json` or nodejs/release data. As such, IMO given the context you've provided here I think either of those names would be more clear.

```suggestion
    data[name].releases[`v${versionSemver.version}`].date = versions[version].release ?? undefined
```

```suggestion
    data[name].releases[`v${versionSemver.version}`].date = versions[version].releaseDate ?? undefined
```

HinataKah0 replied:
Initially I assumed that the field name should be single word 😅 

`releaseDate` sounds good to me, `release` is missing the context IMO.

I think it should be:
```js
data[name].releases[`v${versionSemver.version}`].releaseDate = versions[version].date ?? undefined
```
No?

bnb commented:
I've updated the test data (and fixed some problems with the tooling that regenerates it) in https://github.com/cutenode/nodevu/pull/34. If you want to rebase off https://github.com/cutenode/nodevu/commit/009197937e25a7167e603bf76393086c175dc823 and work on tests from there, you should be good to go.

Generally with tests in this repo I just want to make sure the data propagates through all the potential variants of how someone would use it (like custom URLs) and test the various potential states, even if they're static. Since this data is 100% static and just getting passed through from the source to the module, checking a few known values is enough IMO. Perhaps the earliest known value for each, a random middle value, and then the (current) most recent. Can be / should be hardcoded 👍🏻 

Also: if that's too much, I'm totally happy to tweak the tests to get this over the finish line and unblock y'all.

HinataKah0 commented:
Thank you very much for your kind response!

I've rebased, added into docs, and added tests.

I hope I understood your comment correctly. If I got something wrong, feel free to tweak my commit if there's anything you want to change. Or you can let me know, I'll be more than happy to do it.

ovflowd commented:
Thank you @HinataKah0 for working on this ❤️ (And Tierney for reviewing it 💯)

ovflowd commented:
Hey @bnb is this on a ready-state? 👀

bnb commented:
my apologies, I was at a conference in Romania and have been sick. I'll release now @ovflowd :)

ovflowd commented:
Oh no need to apologise at all! Hope you're feeling better!

And thanks for the merge ✨

HinataKah0 commented:
Thank you!
No worry, take care :)

bnb commented:
published as @nodevu/core@0.1.0. Thank you for the contribution @HinataKah0 ❤️
